src/esp/assets/ResourceManager.cpp

```diff
@@ -1984,6 +1988,119 @@ void ResourceManager::removePrimitiveMesh(int primitiveID) {
   primitive_meshes_.erase(primitiveID);
 }
 
+bool ResourceManager::loadAsset(
```

Comment:
This function is very similar to the existing loadAndCreateRenderAssetInstance. Can you consolidate so we don't have code duplication?